# Introduction

## Central dogma of molecular biology
<img width="964" alt="image" src="https://github.com/HuangShiLab/zhangyf/assets/170502144/78d81876-3272-4032-b3f8-efdea4e42a8b">

We harbor a collection of DNA, RNA, protein, and metabolites from both human and microbial cells. DNA can provide instructions for proteins. During the expression of a protein-coding gene, the central dogma of molecular biology states that genetic information flows only in one direction, from DNA, to RNA, to protein, ultimately affecting the metabolite levels. 

<img width="950" alt="image" src="https://github.com/HuangShiLab/zhangyf/assets/170502144/aaf64e44-5a01-442b-80dd-f19b27cca63b">

While performing microbiome data analysis, we collect microbiome samples and extract DNA for sequencing to uncover the potential capabilities of microbes by asking "Who are they?" and "What can they do?". However, the actual gene expression is reflected by transcription. Transcriptomics analysis involves isolation and purification of RNA from samples, followed by conversion from RNA to cDNA with reverse transcriptase. The cDNA is loaded in a sequencer and differential expressions are analysed.

## Transcriptome data analysis workflow
<img width="965" alt="image" src="https://github.com/HuangShiLab/zhangyf/assets/170502144/0aa4f270-751f-453e-846a-6cf8015b5d64">
The main workflow of transcriptome data analysis contains five steps.

### 1. Quanlity control
1.1 Sequencing data quality analysis  
Software: FastQC v0.11.9  
Inputdata: Raw data (fastq.gz file)  
Commands:  
```
output_dir=${outdir}/fastqc
mkdir -p ${output_dir}
fastqc -o "$output_dir" "$input_file1" "$input_file2"
```

1.2 Data Filtering  
Software: Cutadapt version 4.8  
Method description:  
(1) remove the adapter sequences  
(2) remove the 5' or 3' end bases that contains N’s or of quality values below 20  
(3) remove reads that are less than 75 bp after trimming  
Inputdata: Raw data (fastq.gz file)  
Commands:  
```
output_dir=${outdir}/cutadapt
mkdir -p ${output_dir}
cutadapt -q 20 -e 0.1 -O 1 --minimum-length=75 --max-n=0.1 \
  -a AGATCGGAAGAG -A AGATCGGAAGAG \
  -o ${output_dir}/trimmed_${sample_name}_R1.fastq.gz -p ${output_dir}/trimmed_${sample_name}_R2.fastq.gz \
  ${input_file1} ${input_file2}
```

Notes: Adapter sequences:  
•	Illumina Universal Adapter — AGATCGGAAGAG  
•	Illumina Small RNA 3' Adapter — TGGAATTCTCGG  
•	Illumina Small RNA 5' Adapter — GATCGTCGGACT  
•	Nextera Transposase Sequence — CTGTCTCTTATA
<img width="1170" alt="Screenshot 2024-06-11 at 12 30 14" src="https://github.com/HuangShiLab/zhangyf/assets/170502144/41cf03d1-5803-493f-96b3-c88048434e57">
According the results of fastqc, the samples in this project used the illumina universal adapter.

### 2. Alignment the clean data to the reference genome  
Software: Bowtie2 v2.5.0  
Inputdata: clean data (fastq.gz)  
Commands:  
```
input_dir=${outdir}/cutadapt
output_dir=${outdir}/bowtie2
mkdir -p ${output_dir}
bowtie2 -x ${bowtie2_index} -1 ${input_dir}/trimmed_${sample_name}_R1.fastq.gz -2 ${input_dir}/trimmed_${sample_name}_R2.fastq.gz -S ${output_dir}/${sample_name}.sam -p 10
```
Note:  
In this project, the reference database is as follows:  
Species: Staphylococcus aureus  
Version: NC_007793.1  
Origin: NCBI  
Download link: https://www.ncbi.nlm.nih.gov/nuccore/NC_007793.1[https://www.ncbi.nlm.nih.gov/nuccore/NC_007793.1]

### 3. Expression analysis
3.1  Sort the alignment results from the clean data to the reference genome  
Software: samtools 1.17-25-geb3be52  
Inputdata: sam file generated by bowtie2  
Commands:  
```
input_dir=${outdir}/bowtie2
output_dir=${outdir}/bowtie2
mkdir -p ${output_dir}
samtools view -bS ${input_dir}/${sample_name}.sam | samtools sort -o ${output_dir}/${sample_name}_sorted.bam
```

3.2 Assemble Transcripts  
Software: StringTie v2.2.3  
Method description:  
After alignment, assemble the aligned reads into transcripts. This step converts the alignment data (usually in SAM/BAM format) into a set of transcripts that represent likely gsene models.  
Inputdata: sorted bam file  
Commands:  
```
input_dir=${outdir}/bowtie2
output_dir=${outdir}/stringtie
stringtie -p 8 -o ${output_dir}/${sample_name}_assembled_transcripts.gtf ${input_dir}/${sample_name}_sorted.bam
```

3.3 Calculate Total Exon Fragments
Software: HTSeq 2.0.7  
Inputdata: sorted sam file; assembled transcripts file in gtf format  
Commands:  
```
input_dir1=${outdir}/bowtie2
input_dir2=${outdir}/stringtie
output_dir=${outdir}/gene_expression_count
mkdir -p ${output_dir}
htseq-count -f bam -s no -r pos ${input_dir1}/${sample_name}_sorted.sam ${input_dir2}/${sample_name}_assembled_transcripts.gtf > ${output_dir}/${sample_name}_counts.txt
```

3.4 Count Mapped Reads [Millions]  
Software: samtools 1.17-25-geb3be52  
Inputdata: sorted bam file 
Commands:
```
input_dir=${outdir}/bowtie2
output_dir=${outdir}/mapped_reads_count
mkdir -p ${output_dir}
samtools view -c -F 260 ${input_dir}/${sample_name}_sorted.bam > ${output_dir}/${sample_name}_mapped_reads_count.txt
```

3.5 Calculate Exon Length [KB]  
Inputdata: assembled transcripts file in gtf format  
Commands:
```
input_dir=${outdir}/stringtie
output_dir=${outdir}/exon_length
mkdir -p ${output_dir}
awk '$3 == "exon" {print $1, $4, $5}' ${input_dir}/${sample_name}_assembled_transcripts.gtf | awk '{sum += $3 - $2 + 1} END {print sum/1000}' > ${output_dir}/${sample_name}_exon_length.txt
```

3.6 Combine count of total exon fragments, count of mapped reads, and exon length for FPKM Calculation
<img width="421" alt="Screenshot 2024-06-13 at 15 05 56" src="https://github.com/HuangShiLab/zhangyf/assets/170502144/ede22a26-9612-4cbd-9170-df70fe52c606">
